### ISSUE TITLE
How to import lookup history to Anki?

### ISSUE THREAD

**Issue by abdrau26** [koreader/koreader#6502]
Is it possible to import lookup history into csv file and then to Anki?

**Comment by pazos** [comment]
> Is it possible to export lookup history

Nope. You need to do it manually. Lookup history is located in `settings/lookup_history.lua `

 It could make sense in the future to provide a way to export/import settings. Not exactly to/from csv but PR for that are welcome.

> and then to Anki?

That's not our stuff. Cannot be fixed/supported by us

**Comment by pazos** [comment]
@abdrau26: if you have an specific workflow in mind, please tell, because integration with 3rd party apps is mostly welcomed. I don't know a thing about Anki :p  

Please do some research first:

- has the app ways to import content from files?
- which type of content and format?. Providing a working example would be great (from OP: a csv file with an anki flashcard or whatever source format anki uses to make a flashcard)

But most important: please detail the workflow you're trying to achieve. Do you want to memoize your lookup history? :)

**Comment by abdrau26** [comment]
1. I want to memorize my lookup history
2. Conversion of lookup_history.lua to csv format. 
3. It would be better if csv file contains word, sentence and meaning in three columns. But only word column will suffice my need.
4. Anki supports import from csv file in its PC version software.

**Comment by abdrau26** [comment]
> 1. I want to memorize my lookup history
> 2. Conversion of lookup_history.lua to csv format.
> 3. It would be better if csv file contains word, sentence and meaning in three columns. But only word column will suffice my need.
> 4. Anki supports import from csv file in its PC version software.

@pazos

**Comment by pazos** [comment]
> It would be better if csv file contains word, sentence and meaning in three columns. But only word column will suffice my need.

the context setence is not available right now but the rest is doable as-is. 

@abdrau26: please provide a sample `lookup_history.lua` and a sample csv with a couple of entries extracted from the lookup history that make the flashcards you want.

**Comment by abdrau26** [comment]
First 5 entries of lookup_history.lua and csv sample file:
.lua file
https://drive.google.com/file/d/11RZxiMXi9vduASfG48OfGXlvwd-AlAc6/view?usp=drivesdk
.csv file
https://drive.google.com/file/d/1K1dE_QrMFhLl1wTJs92h05yuxGzoXmKd/view?usp=drivesdk

**Comment by pazos** [comment]
@abdrau26: thanks. You attached the whole settings, not the lookup_history :p 

If I understood correctly you want the csv with the definition on the 2nd column. Also, how your sample is rendered by anki? One row per flashcard?. A screenshot of the sample csv rendered by anki would be great too.

**Comment by abdrau26** [comment]
What will happen if I use 5 dictionaries? Will csv file contain 6 columns (1 for word and 5 for meanings)?
Let us say that only words are imported in csv file.
Then i will import csv to Anki.
Anki flashcard for a row will look like this:
![Screenshot_2020-08-15_234850](https://user-images.githubusercontent.com/59373108/90319480-07d60400-df52-11ea-96a0-d8997dc3f353.jpg)
Words will go to front of the card. I will manually add back  of the card which will be definition(Add reverse is not relevent).
If there are two columns in the csv file, first column represents front of the card and second column represents back of the card.
One row in csv represents one card in Anki.
If there are more than two columns in a csv file, I don't know what will happen to Anki import.

**Comment by abdrau26** [comment]
Csv file is imported by Anki. Anki can not export csv file.

**Comment by pazos** [comment]
> What will happen if I use 5 dictionaries? Will csv file contain 6 columns (1 for word and 5 for meanings)?

That sounds pretty bad. Actually I was trying to understand how anki csv works without installing Anki and messing with it myself.

My suggestion would be running sdcv (ie: retrieving dict definition) for each entry in the lookup history, so it can become slow with a large number of entries with a single definition, much much more if you run sdcv against each dictionary installed.

**Comment by anatoly314** [comment]
Don't need to mess with ANKI, as @pazos said it isn't related to Koreader at all. Moreover, it isn't enough to import it to ANKI it needs to be translated, formatted, etc...  I've done a POC project to take a list of words => translate them => create ANKI cards => import to ANKI here: https://github.com/anatoly314/data2anki , you can see it as a reference but project is considered deprecated right now, and I will write it from scratch.

 But to get an easy way to export lookup words (include whole sentences which were translated by google translate) it will be great. Actually providing `lookup_history.lua` is a good beginning.

**Comment by mallomar** [comment]
Any progress on this? I'm moving from Kindle to KOReader on an Onyx and I can export my Kindle Vocabulary Builder to CSV and it would be great if I could do the same with KOReader dictionary lookup history. I'm also importing the words to Anki.
